lib/rubocop/cop/style/file_read.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Favor `File.(bin)read` convenience methods.
+      #
+      # @example text mode
+      #   # bad
+      #   File.open(filename).read
+      #   File.open(filename, &:read)
+      #   File.open(filename) { |f| f.read }
+      #   File.open(filename) do |f|
+      #     f.read
+      #   end
+      #   File.open(filename, 'r').read
+      #   File.open(filename, 'r', &:read)
+      #   File.open(filename, 'r') { |f| f.read }
+      #   File.open(filename, 'r') do |f|
+      #     f.read
+      #   end
```

Comment:
These two examples are fundamentally the same so we don't really need both (same for lines 12-15)